// @pulumi/digitalocean/genaiFunction.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.GenaiFunction = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
class GenaiFunction extends pulumi.CustomResource {
    /**
     * Get an existing GenaiFunction resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new GenaiFunction(name, state, Object.assign(Object.assign({}, opts), { id: id }));
    }
    /**
     * Returns true if the given object is an instance of GenaiFunction.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === GenaiFunction.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["agentId"] = state ? state.agentId : undefined;
            resourceInputs["description"] = state ? state.description : undefined;
            resourceInputs["faasName"] = state ? state.faasName : undefined;
            resourceInputs["faasNamespace"] = state ? state.faasNamespace : undefined;
            resourceInputs["functionName"] = state ? state.functionName : undefined;
            resourceInputs["functionUuid"] = state ? state.functionUuid : undefined;
            resourceInputs["inputSchema"] = state ? state.inputSchema : undefined;
            resourceInputs["outputSchema"] = state ? state.outputSchema : undefined;
        }
        else {
            const args = argsOrState;
            if ((!args || args.agentId === undefined) && !opts.urn) {
                throw new Error("Missing required property 'agentId'");
            }
            if ((!args || args.description === undefined) && !opts.urn) {
                throw new Error("Missing required property 'description'");
            }
            if ((!args || args.faasNamespace === undefined) && !opts.urn) {
                throw new Error("Missing required property 'faasNamespace'");
            }
            if ((!args || args.functionName === undefined) && !opts.urn) {
                throw new Error("Missing required property 'functionName'");
            }
            if ((!args || args.inputSchema === undefined) && !opts.urn) {
                throw new Error("Missing required property 'inputSchema'");
            }
            resourceInputs["agentId"] = args ? args.agentId : undefined;
            resourceInputs["description"] = args ? args.description : undefined;
            resourceInputs["faasName"] = args ? args.faasName : undefined;
            resourceInputs["faasNamespace"] = args ? args.faasNamespace : undefined;
            resourceInputs["functionName"] = args ? args.functionName : undefined;
            resourceInputs["inputSchema"] = args ? args.inputSchema : undefined;
            resourceInputs["outputSchema"] = args ? args.outputSchema : undefined;
            resourceInputs["functionUuid"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(GenaiFunction.__pulumiType, name, resourceInputs, opts);
    }
}
exports.GenaiFunction = GenaiFunction;
/** @internal */
GenaiFunction.__pulumiType = 'digitalocean:index/genaiFunction:GenaiFunction';
